r2i/engine.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include "engine.h"
+
+namespace r2i {
+
+RuntimeError Engine::SetModel (std::shared_ptr<r2i::IModel> in_model) {
+  RuntimeError error;
+  error.Set(RuntimeError::NOT_IMPLEMENTED, "Engine::SetModel method not implemented");
+  return error;
+}
+
+RuntimeError Engine::Start () {
+  RuntimeError error;
+  error.Set(RuntimeError::NOT_IMPLEMENTED, "Engine::Start method not implemented");
+  return error;
+}
+
+RuntimeError Engine::Stop () {
+  RuntimeError error;
+  error.Set(RuntimeError::NOT_IMPLEMENTED, "Engine::Stop method not implemented");
+  return error;
+}
+
+std::shared_ptr<r2i::IPrediction> Engine::Predict (std::shared_ptr<r2i::IFrame>
```

Comment:
I don't like the fact that we are letting subclasses remember to call the pre and post process. We should absorb this common logic here somehow.